### PR TITLE
Update gradle deps and clean up code with Lint

### PR DESCRIPTION
**Pull request by IgorGanapolsky**
This commit updates to the latest gradle dependencies and follows the latest Gradle plugin DSL in Android Studio.
It also does a code cleanup as per Lint recommendations.

### CODE REVIEW

File: device/iot-device-samples/android-sample/app/src/main/java/com/iothub/azure/microsoft/com/androidsample/MainActivity.java [hunk 1]
```diff
@@ -23,19 +29,12 @@ protected void onCreate(Bundle savedInstanceState) {
...
-        catch(Exception e2)
-        {
+        } catch (Exception e2) {
             System.out.println("Exception while opening IoTHub connection: " + e2.toString());
         }
     }
 
-    public void SendMessage() throws URISyntaxException, IOException
-    {
+    private void SendMessage() throws URISyntaxException, IOException {
```
Comment (prmathur-microsoft) [inline device/iot-device-samples/android-sample/app/src/main/java/com/iothub/azure/microsoft/com/androidsample/MainActivity.java]: As per our coding guidelines we start braces in next line. 
Refer to documentation - https://github.com/Azure/azure-iot-sdk-java/blob/master/.github/CODING_GUIDELINES.md#space-and-braces

Comment (IgorGanapolsky) [reply device/iot-device-samples/android-sample/app/src/main/java/com/iothub/azure/microsoft/com/androidsample/MainActivity.java]: Out of curiosity, why aren't you following IntelliJ's and most Java developers' code conventions?

Comment (prmathur-microsoft) [reply device/iot-device-samples/android-sample/app/src/main/java/com/iothub/azure/microsoft/com/androidsample/MainActivity.java]: We have gathered our coding conventions from various sources across the web coding in Java and this document shows a compilation of them. There are lot of defaults in intelliJ which do not abide by guidelines and a developer will have to change those settings to follow CG. You have encountered one such change here.

**Comment by prmathur-microsoft** [comment]
This PR has been rebased to master, and commits are squashed into single commit and merge.

**Pull request closed without merging**